src/ExactDiagonalization/operator_as_map.jl

```diff
@@ -0,0 +1,104 @@
+struct OperatorAsMap{T,H<:AbstractOperator{T},A} <: LinearMap{T}
```

Comment:
I'd say it deserves a docstring. Maybe a brief one that identifies it as an internal struct created by `LinearOperator` and referring to its documentation?